src/Microsoft.Identity.Client/TokenCacheExtensions.cs

```diff
@@ -100,8 +106,70 @@ public static void Deserialize(this TokenCache tokenCache, byte[] unifiedState)
             GuardOnMobilePlatforms();
             lock (tokenCache.LockObject)
             {
-                RequestContext requestContext = new RequestContext(null, new MsalLogger(Guid.Empty, null));
-                TokenCacheSerializeHelper.DeserializeUnifiedCache(tokenCache.TokenCacheAccessor, unifiedState, requestContext);
+                var requestContext = tokenCache.CreateRequestContext();
+                tokenCache.TokenCacheAccessor.Clear();
+
+                Dictionary<string, IEnumerable<string>> cacheDict = JsonHelper
+                    .DeserializeFromJson<Dictionary<string, IEnumerable<string>>>(unifiedState);
+
+                if (cacheDict == null || cacheDict.Count == 0)
+                {
+                    tokenCache.Logger.Info("Msal Cache is empty.");
```

Comment:
>tokenCache.Logger [](start = 20, length = 17)

slightly odd to me that the logger hangs on the tokenCache obj, though I see why....

Reply:
Yes, it is odd, and it's why ICoreLogger is an internal there.  We have a TON of static helpers, extensions, etc that should just honestly be on the TokenCache class itself instead of being spread around.  I cleaned up a bit of this with one-use functions and I'd like to look at doing it with the Extensions here because I'm not sure why they're needed to be extensions instead of just on the class itself.  That would clean this up and allow the logger to just be a private method on the tokencache class.